test/run_test.py

```diff
@@ -435,20 +435,43 @@ def pull_job_times_from_S3() -> Dict[str, Tuple[float, int]]:
     return calculate_job_times(s3_reports)
 
 
+def get_S3_job_times() -> Dict[str, float]:
+    if os.path.exists(TEST_TIMES_FILE):
+        with open(TEST_TIMES_FILE) as file:
+            test_times_json: JobTimeJSON = json.load(file)
+
+        curr_commit = subprocess.check_output(['git', 'rev-parse', 'HEAD'], encoding="ascii").strip()
```

Comment:
would this work if my local git is dirty?
It occurs to me that most of the S3 improvement has assumption that it is running in CI. should we guard the historic test time based sharding with `IN_CI` ?

Reply:
Oh hm this works and just uses the last commit (disregarding uncommitted changes). I don't think it matters that much because the stats are pulled based on the last commit anyway.